include/deal.II/lac/constraint_matrix.h

```diff
@@ -1406,6 +1421,8 @@ ConstraintMatrix::add_entry (const size_type line,
   // exists, since we don't want to enter it twice
   Assert (lines_cache[calculate_line_index(line)] != numbers::invalid_size_type,
           ExcInternalError());
+  Assert (!local_lines.size() || local_lines.is_element(column),
```

Comment:
> @bangerth <https://github.com/bangerth>: I actually don't see how DoFs that
> extend to the other side of the domain can pose any problems. They are locally
> relevant if add_periodicity is used.

Ah, indeed. Good point. Thanks for clearing it up!

> The only problem that I can think of is if
> 1.) the ConstraintMatrix is initialized with an IndexSet describing the
> locally active DoFs and 2.) constraints that reference locally relevant DoFs
> not in this IndexSet are used.
> Basically, one is forced to initialize the ConstraintMatrix with the locally
> relevant IndexSet. Is this too restrictive?

No, I think that's exactly what needs to be done.